src/engine/test/assets/decoders/cisco-asa.yml

```diff
@@ -0,0 +1,198 @@
+
+    # %ASA-4-113019: Group = 81.2.69.143, Username = 81.2.69.143, IP = 81.2.69.143, Session disconnected. Session Type: LAN-to-LAN, Duration: 0h:32m:16s, Bytes xmt: 297103, Bytes rcv: 1216163, Reason: User Requested
+    # %ASA-4-113019: Group = TheBeatles, Username = Ringo, IP = 234.56.12.87, Session disconnected. Session Type: AnyConnect-Parent, Duration: 0h:01m:52s, Bytes xmt: 32452, Bytes rcv: 0, Reason: User Requested
+    # %ASA-4-113019: Group = TheBeatles, Username = John, IP = 234.28.45.42, Session disconnected. Session Type: SSL, Duration: 2h:27m:34s, Bytes xmt: 45323434, Bytes rcv: 43252324, Reason: Idle Timeout
+    - message:  >-
+        Group = <source.user.group.name>, Username = <source.user.name>, IP = <destination.address>,
+        Session disconnected. Session Type: <_cisco.session_type>, Duration: <_duration_hms>, Bytes xmt: <source.bytes>,
+        Bytes rcv: <destination.bytes>, Reason: <event.reason>
+
+
+    # %ASA-6-302013: Built outbound TCP connection 7827084 for INSIDE:192.168.100.165/443 (192.168.100.165/443) to OUTSIDE:173.43.124.164/3465 (173.43.124.164/3465)
+    # %ASA-6-302013: Built outbound TCP connection 7827084 for INSIDE:192.168.100.165/443 (192.168.100.165/443) to OUTSIDE:173.43.124.164/3465 (173.43.124.164/3465)
+    # AND
+    # %ASA-6-302015: Built inbound UDP connection 1556719 for inside:10.10.0.21/22783 (10.10.0.21/22783) to identity:89.123.53.12/4500 (89.123.53.12/4500)
+    # %ASA-6-302015: Built inbound UDP connection 1556719 for inside:10.10.0.21/22783 (10.10.0.21/22783) to identity:89.123.53.12/4500 (89.123.53.12/4500)
+    # %ASA-6-302015: Built inbound UDP connection 1556719 for inside:10.10.0.21/22783 (10.10.0.21/22783) to identity:89.123.53.12/4500 (89.123.53.12/4500)
+    - message:  >-
+        Built <network.direction> <network.protocol> connection <_cisco.conn.id> for
+        <observer.egress.interface.name>:<source.ip>/<source.port> (<source.nat.ip>/<source.nat.port>) to
+        <observer.egress.interface.name>:<source.ip>/<source.port> (<source.nat.ip>/<source.nat.port>) <destination.user.name>
+    - message:  >-
+        Built <network.direction> <network.protocol> connection <_cisco.conn.id> for
+        <observer.egress.interface.name>:<source.ip>/<source.port> (<source.nat.ip>/<source.nat.port>) to
+        <observer.egress.interface.name>:<source.ip>/<source.port> (<source.nat.ip>/<source.nat.port>)
+
+normalize:
+  - map:
+      observer.product: ASA
+      observer.type: firewall
+      cisco.asa.message_id: $event.code
+      event.action: firewall-rule
+      event.category: [ network ]
+      event.type: [ connection, firewall ]
+      # Only one of the following will be set
+      source.ip: $source.address
+      source.host: $source.address
+      # Only one of the following will be set
+      destination.ip: $destination.address
+      destination.host: $destination.address
+      related.ip: +s_append/$source.ip
+      related.ip: +s_append/$destination.ip
```

Comment:
At the moment, the same field cannot be mapped two times under the same "map" block

Reply:
Done!